server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -378,10 +381,10 @@ public ActionRequestValidationException validate() {
                         validationException
                     );
                 }
-                if (source.knnSearch().isEmpty() || source.query() == null && source.knnSearch().size() < 2) {
+                int queryCount = source.subSearches().size() + source.knnSearch().size();
+                if (queryCount < 2) {
                     validationException = addValidationError(
-                        "[rank] requires a minimum of [2] result sets which"
-                            + " either needs at minimum [a query and a knn search] or [multiple knn searches]",
+                        "[rank] requires a minimum of [2] result sets using a combination of queries and/or knn searches",
```

Comment:
now that its called `sub_searches` you need to update this error I think.

Reply:
Fixed.